src/libs/actions/Report.js

```diff
@@ -1213,45 +1247,89 @@ function editReportComment(reportID, originalReportAction, textForNewComment) {
 
     // Do not autolink if someone explicitly tries to remove a link from message.
     // https://github.com/Expensify/App/issues/9090
-    const htmlForNewComment = parser.replace(textForNewComment, {filterRules: _.filter(_.pluck(parser.rules, 'name'), name => name !== 'autolink')});
+    const htmlForEditedComment = parser.replace(textForNewComment, {filterRules: _.filter(_.pluck(parser.rules, 'name'), name => name !== 'autolink')});
+    const textForEditedComment = parser.htmlToText(htmlForEditedComment);
 
     //  Delete the comment if it's empty
-    if (_.isEmpty(htmlForNewComment)) {
+    if (_.isEmpty(htmlForEditedComment)) {
         deleteReportComment(reportID, originalReportAction);
         return;
     }
 
     // Skip the Edit if message is not changed
-    if (originalReportAction.message[0].html === htmlForNewComment.trim()) {
+    if (originalReportAction.message[0].html === htmlForEditedComment.trim()) {
         return;
     }
 
-    // Optimistically update the report action with the new message
+    // Optimistically update the reportAction with the new message
     const sequenceNumber = originalReportAction.sequenceNumber;
-    const newReportAction = {...originalReportAction};
-    const actionToMerge = {};
-    newReportAction.message[0].isEdited = true;
-    newReportAction.message[0].html = htmlForNewComment;
-    newReportAction.message[0].text = parser.htmlToText(htmlForNewComment);
-    actionToMerge[sequenceNumber] = newReportAction;
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionToMerge);
-
-    // Persist the updated report comment
-    DeprecatedAPI.Report_EditComment({
+    const optimisticReportActions = {
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+            message: [{
+                isEdited: true,
+                html: htmlForEditedComment,
+                text: textForEditedComment,
+            }],
+        },
+    };
+
+    // Optimistically update the report's last message if the user edited it
+    const highestSequenceNumber = getMaxSequenceNumber(reportID);
+    const optimisticReport = {};
+    if (sequenceNumber === highestSequenceNumber) {
+        optimisticReport.lastMessageText = ReportUtils.formatReportLastMessageText(textForEditedComment);
+    }
+
+    // List of optmistic data to change
```

Comment:
```suggestion
```

**tip:** try to avoid comments that describe what the code does without explaining why